### PR TITLE
Enforce Code Style on build

### PR DESCRIPTION
**Pull request by lesaltzm**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -41,34 +41,34 @@ dotnet_sort_system_directives_first = true
 dotnet_separate_import_directive_groups = false
```
Comment (lesaltzm) [inline .editorconfig:42]: This file matters

File: src/PowerFx.ruleset [hunk 2]
```diff
@@ -190,8 +173,30 @@
     <Rule Id="SA1642" Action="Error" />
     <Rule Id="SA1643" Action="Error" />
     <Rule Id="SA1648" Action="Error" />
-    <Rule Id="SA1649" Action="Error" />
     <Rule Id="SA1651" Action="Error" />
     <Rule Id="SX1309" Action="Error" /> <!-- FieldNamesShouldBeginWithUnderscores -->
+    <!-- XML Documentation Warnings, disabled until we take a pass over docs -->
```
Comment (lesaltzm) [inline src/PowerFx.ruleset:178]: And this

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/CommentToken.cs [hunk 3]
```diff
@@ -8,17 +8,16 @@ namespace Microsoft.PowerFx.Core.Lexer.Tokens
...
     internal class CommentToken : Token
     {
-        private readonly string _val;
         public bool IsOpenBlock;
 
         public CommentToken(string val, Span span)
             : base(TokKind.Comment, span)
         {
             Contracts.AssertValue(val);
-            _val = val;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/CommentToken.cs:18]: >_val [](http://example.com/codeflow?start=12&length=4)

Why this change?
Is there something wrong with a private field?

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/CommentToken.cs:18]: auto applying https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0032

Nothing wrong with a private field, but if we're adding style rules, we might as well have consistent application of them.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/CommentToken.cs:18]: Can you clarify what "consistent application" means.  Eg, do the new style rules discourage private fields?

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/CommentToken.cs:18]: That style rule discourages using a private field that's just a backing for a public property, yes.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]
:shipit:
